crates/turbopack-convert-trace/src/main.rs

```diff
@@ -411,6 +430,95 @@ fn main() {
             match task {
                 Task::Enter { id, root } => {
                     let mut span = take(&mut spans[id]);
+                    let mut count = span.count;
+                    let mut items = take(&mut span.items);
+                    if graph && !items.is_empty() {
+                        let parent_name = &*span.name;
+                        let mut groups = IndexMap::new();
+                        let mut self_items = Vec::new();
+                        fn add_items_to_groups(
+                            groups: &mut IndexMap<&str, Vec<SpanItem>>,
+                            self_items: &mut Vec<SpanItem>,
+                            spans: &mut Vec<Span>,
+                            parent_count: &mut u32,
+                            parent_name: &str,
+                            items: Vec<SpanItem>,
+                        ) {
+                            for item in items {
+                                match item {
+                                    SpanItem::SelfTime { .. } => {
+                                        self_items.push(item);
+                                    }
+                                    SpanItem::Child(id) => {
+                                        // SAFETY: We never mutate the `name` of the spans or the
+                                        // Vec itself. We only mutate the `items` Vec.
+                                        let key = unsafe { &*(&*spans[id].name as *const str) };
```

Comment:
Do we really need to use unsafe here? Is there a measurable performance benefit? Otherwise, let's keep it to safe Rust only.

Reply:
Rust will complain about mutable borrow otherwise, since we `take(&mut spans[id].items)` later. So we can't keep a immutable borrow on `span[id].name` while mutating `spans[id].items`.

The alternative would be to clone the name.

Reply:
Is there a performance issue with cloning the name, or using an `Arc<String>` here?

Reply:
I measured no difference when using clone instead. So it's now safe.